velox/serializers/tests/PrestoSerializerTest.cpp

```diff
@@ -1511,34 +1511,14 @@ INSTANTIATE_TEST_SUITE_P(
         common::CompressionKind::CompressionKind_LZ4,
         common::CompressionKind::CompressionKind_GZIP));
 
-TEST_F(PrestoSerializerTest, deserializeSingleColumn) {
-  // Verify that deserializeSingleColumn API can handle all supported types.
-  static const size_t kPrestoPageHeaderBytes = 21;
-  static const size_t kNumOfColumnsSerializedBytes = sizeof(int32_t);
-  static const size_t kBytesToTrim =
-      kPrestoPageHeaderBytes + kNumOfColumnsSerializedBytes;
-
+TEST_F(PrestoSerializerTest, serdeSingleColumn) {
+  // Verify that (de)serializeSingleColumn API can handle all supported types.
   auto testRoundTripSingleColumn = [&](const VectorPtr& vector) {
-    auto rowVector = makeRowVector({vector});
-    // Serialize to PrestoPage format.
     std::ostringstream output;
-    auto arena = std::make_unique<StreamArena>(pool_.get());
-    auto rowType = asRowType(rowVector->type());
-    auto numRows = rowVector->size();
-    auto serializer = serde_->createIterativeSerializer(
-        rowType, numRows, arena.get(), nullptr);
-    serializer->append(rowVector);
-    facebook::velox::serializer::presto::PrestoOutputStreamListener listener;
-    OStreamOutputStream out(&output, &listener);
-    serializer->flush(&out);
-
-    validateLexer(output.str(), {});
+    serde_->serializeSingleColumn(vector, nullptr, pool_.get(), &output);
+    const auto serialized = output.str();
 
-    // Remove the PrestoPage header and Number of columns section from the
-    // serialized data.
-    std::string input = output.str().substr(kBytesToTrim);
-
-    auto byteStream = toByteStream(input);
+    auto byteStream = toByteStream(serialized);
```

Comment:
Why did you remove the validateLexer call ?

Reply:
`validateLexer` call was removed here because the `PrestoPage` header is no longer present in the serialized data obtained from the call to `serializeSingleColumn`, and this results in the following error from `validateLexer`:
```
velox/velox/serializers/tests/PrestoSerializerTest.cpp:177: Failure
Value of: status.ok()
  Actual: false
Expected: true
PrestoPage header invalid
```
This wasn't an issue before because the function `serialize` was used to obtain the serialized data including the `PrestoPage` header, and the header bytes were just skipped over to test the function `deserializeSingleColumn`. Please let me know if this is fine.

Reply:
Then it might be better to test both code paths (the original one with serialize and your new one using serializeColumn) separately, so that the distinction is clear. Lets not completely overwrite the original test.

Reply:
Thanks for the suggestion @aditi-pandit, added another lambda, `testSerializeSingleColumnRoundTrip`, to test `serializeSingleColumn` and `deserializeSingleColumn` round trip without modifying the existing code path. Could you please take another look?